module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.cpp

```diff
@@ -385,7 +386,7 @@ namespace behaviour {
                     timeLastObjectSeen = now;
                     std::set<Goal::Side> visiblePosts;
                     // TODO treat goals as one object
-                    std::vector<Goal> goals;
+                    Goals goals;
```

Comment:
Basically throughout the rest of this file, go through and only have the Balls or Goals objects when they come in from an on statement, or when they are going to be emitted.